crypto/fipsmodule/rand/new_rand_test.cc

```diff
@@ -37,6 +38,26 @@ TEST(NewRand, Basic) {
     ASSERT_TRUE(RAND_bytes_with_additional_data(randomness, i, user_personalization_string));
     ASSERT_TRUE(RAND_bytes_with_user_prediction_resistance(randomness, i, user_personalization_string));    
   }
+
+  *returnFlag = true;
+}
+
+TEST(NewRand, Basic) {
```

Comment:
This is a reasonable test for the happy case, can you add a test for the new cleanup code that ensures the locking works as expected. You might need to migrate this test from threads to futures which supports timeouts.